migrations/versions/10_adding_supplier_to_user_table.py

```diff
@@ -0,0 +1,36 @@
+"""empty message
+
+Revision ID: 10_adding_supplier_to_user_table
+Revises: 3d5aabf7d291
+Create Date: 2015-04-21 07:45:29.958447
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '10_adding_supplier_to_user_table'
+down_revision = '3d5aabf7d291'
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    op.add_column('users', sa.Column('supplier_id', sa.BigInteger(), nullable=True))
+    op.create_index(op.f('ix_users_supplier_id'), 'users', ['supplier_id'], unique=False)
+    op.create_foreign_key(None, 'users', 'suppliers', ['supplier_id'], ['supplier_id'])
+
+    op.create_check_constraint(
+        "ck_user_supplier_has_supplier_id",
+        "users",
+        "((role='buyer') or (role='admin') or (role = 'supplier'  and supplier_id is not null))"
```

Comment:
As discussed in real life: We're going to add the check constraint to the SQLAlchemy model so that it's more obvious to people developing against that model. With an eye to replacing this field with the [enum recipe](http://techspot.zzzeek.org/2011/01/14/the-enum-recipe/) in the future